skimage/transform/steerable.py

```diff
@@ -0,0 +1,425 @@
+            lo0dft, log_rad, angle, Xrcos, Yrcos, self.height - 1)
+
+        hi0dft = imdft * hi0mask
+        hi0 = np.fft.ifft2(np.fft.ifftshift(hi0dft))
+
+        coeff.insert(0, [hi0.real])
+
+        return coeff
+
+    def _build_pyr_level(self, lodft, log_rad, angle, Xrcos, Yrcos, ht):
+        """ Recursively calculate Steerable subbands
+
+        Parameters
+        ----------
+        lodft : 2-D array
+            DFT matrix of the higher layer
+        log_rad, angle: 2-D array
+            Helper matrices to help create the DFT mask
+        Xrcos, Yrcos: 2-D array
+            Represents the desired filter
+        ht: int
+            Current level of the pyramid that are being built
+
+        Notes
+        -----
+        This method is called recursively until ht reaches 1.
+        """
+        if (ht <= 1):
+            lo0 = np.fft.ifft2(np.fft.ifftshift(lodft))
+            coeff = [[lo0.real]]
+
+        else:
+            Xrcos = Xrcos - 1
+
+            # ==================== Orientation bandpass =======================
+            himask = _point_op(log_rad, Yrcos, Xrcos)
+
+            lutsize = 1024
+            Xcosn = np.pi * \
+                np.array(range(-(2 * lutsize + 1), (lutsize + 2))) / lutsize
+            order = self.n_bands - 1
+            const = np.power(2, 2 * order) * \
+                np.square(np.math.factorial(order)) / \
+                (self.n_bands * np.math.factorial(2 * order))
+
+            alpha = (Xcosn + np.pi) % (2 * np.pi) - np.pi
+            Ycosn = 2 * np.sqrt(const) * \
+                np.power(np.cos(Xcosn), order) * (np.abs(alpha) < np.pi / 2)
+
+            orients = []
+
+            for b in range(self.n_bands):
+                anglemask = _point_op(
+                    angle, Ycosn, Xcosn + np.pi * b / self.n_bands)
+                banddft = np.power(np.complex(
+                    0, -1), self.n_bands - 1) * lodft * anglemask * himask
+                band = np.fft.ifft2(np.fft.ifftshift(banddft))
+                orients.append(band)
+
+            # ================== Subsample lowpass ============================
+            dims = np.array(lodft.shape)
+
+            lostart = np.ceil((dims + 0.5) / 2) - \
+                np.ceil((np.ceil((dims - 0.5) / 2) + 0.5) / 2)
+            loend = lostart + np.ceil((dims - 0.5) / 2)
+
+            lostart = lostart.astype(np.int32)
+            loend = loend.astype(np.int32)
+
+            log_rad = log_rad[lostart[0]:loend[0], lostart[1]:loend[1]]
+            angle = angle[lostart[0]:loend[0], lostart[1]:loend[1]]
+            lodft = lodft[lostart[0]:loend[0], lostart[1]:loend[1]]
+
+            YIrcos = np.abs(np.sqrt(1 - Yrcos * Yrcos))
+            lomask = _point_op(log_rad, YIrcos, Xrcos)
+
+            lodft = lomask * lodft
+
+            coeff = self._build_pyr_level(
+                lodft, log_rad, angle, Xrcos, Yrcos, ht - 1)
+            coeff.insert(0, orients)
```

Comment:
Inserting into a Python list is much more expensive than appending. I suggest you build the pyramid in reverse order, and then at the top level, return `list(reversed(coeff))`